src/topology/tplg_local.h

```diff
@@ -47,6 +47,7 @@ typedef enum _snd_pcm_rates {
 	SND_PCM_RATE_11025,
 	SND_PCM_RATE_16000,
 	SND_PCM_RATE_22050,
+	SND_PCM_RATE_24000,
```

Comment:
is this backwards-compatible @yongzhi1 ?

Reply:
Hi, @plbossart you raised a great question that I have not thought about.

cc @kv2019i @ujfalusi

Reply:
It seems the rates are only used to build topology:
https://github.com/alsa-project/alsa-lib/commit/c550a421a3ff5b1

I think it's safe to add a new one.

Reply:
what happens if you create a topology file with the new rate, and feed it to an older device? Would it try to configure the 32kHz then?

Reply:
yes, that's what happened exactly, posted a kernel side patch https://github.com/thesofproject/linux/pull/4966.

Reply:
I don't understand how you can manage something that depends on alignment of kernel+userspace libs. You will break lots of platforms if one of the two is updated separately.

Reply:
Murphy's law .. open to any suggestion, thanks!!

Reply:
I believe the non defined rates can be accessed if the driver advertises `SNDRV_PCM_RATE_CONTINUOUS` and/or `SNDRV_PCM_RATE_KNOT` and uses constraints, rules to tune the selection.

Reply:
yeah, I remember now I tried to do the same and add a new frequency for HDaudio and @tiwai told me to use the RATE_CONTINUOUS and constraints.